spec/index.adoc

```diff
@@ -2829,9 +2829,8 @@ Conditions::
     Available = S.call_contexts[M.call_contexts].available_cycles
     ( M.entry_point = PublicMethod Name Caller Data
       Arg = { data = Data; caller = Caller }
-      (F = Mod.update_methods[Name](Arg, Env, Available)
-      or
-      (F = query_as_update(Mod.query_methods[Name], Arg, Env))
+      (F = Mod.update_methods[Name](Arg, Env, Available)) or (F = query_as_update(Mod.query_methods[Name], Arg, Env))
+      Name ∉ dom(Mod.update_methods) or Name ∉ dom(Mod.query_methods)
```

Comment:
I was wondering whether this is the right place to put the condition. It is definitely true, but it seems that's kind-of an invariant for all wasm modules and should be true after installation.

Reply:
I agree. It could move to the invariants section and be established by canister creation and upgrade. Let me give it a try!

Reply:
Done!